src/middlewared/middlewared/plugins/account.py

```diff
@@ -287,6 +289,84 @@ async def query(self, filters, options):
             filter_list, result, filters, options
         )
 
+    @private
+    def validate_homedir_path(self, verrors, schema, data, users):
+        needs_additional_validation = True
+        p = Path(data['home'])
+
+        if not p.is_absolute():
+            verrors.add(f'{schema}.home', '"Home Directory" must be an absolute path.')
+            return False
+
+        if p.is_file():
+            verrors.add(f'{schema}.home', '"Home Directory" cannot be a file.')
+            return False
+
+        if ':' in data['home']:
+            verrors.add(f'{schema}.home', '"Home Directory" cannot contain colons (:).')
+            return False
+
+        if data['home'] == DEFAULT_HOME_PATH:
+            return False
+
+        if not p.exists():
+            if data.get('home_create', False):
+                verrors.add(
+                    f'{schema}.home',
+                    f'{data["home"]}: path specified to use for home directory creation does not '
+                    'exist. TrueNAS uses the provided path as the parent directory of the '
+                    'of the newly-created home directory.'
+                )
+
+            else:
+                verrors.add(
+                    f'{schema}.home',
+                    f'{data["home"]}: path specified to use as home directory does not exist.'
+                )
+
+            return False
+
+        in_use = filter_list(users, [('home', '=', data['home'])])
+        if in_use:
+            verrors.add(
+                f'{schema}.home',
+                f'{data["home"]}: homedir already used by {in_use[0]["username"]}.',
+                errno.EEXIST
+            )
+            needs_additional_validation = False
+
+        if not data['home'].startswith('/mnt/'):
+            verrors.add(
+                f'{schema}.home',
+                '"Home Directory" must begin with /mnt/ or set to '
+                f'{DEFAULT_HOME_PATH}.'
+            )
+            needs_additional_validation = False
+        elif not any(
+            data['home'] == i['path'] or data['home'].startswith(i['path'] + '/')
+            for i in self.middleware.call_sync('pool.query')
```

Comment:
Probably want to do something like this instead of calling `pool.query` since it can be particularly expensive (last time I checked)
```
elif not any(
    data['home'] == f'/mnt/{v["name"]}' or data['home'].startswith(f'/mnt/{v["name"]}/'
    for k, v in self.middleware.call_sync('zfs.pool.query_imported_fast').items()
)

Reply:
Yeah, I felt yucky shifting this validation :)